crates/red_knot_python_semantic/src/semantic_index/definition.rs

```diff
@@ -50,6 +50,18 @@ impl<'db> Definition<'db> {
     pub(crate) fn scope(self, db: &'db dyn Db) -> ScopeId<'db> {
         self.file_scope(db).to_scope_id(db, self.file(db))
     }
+
+    /// Returns the definition's kind which gives access to its AST.
+    ///
+    /// `query_file` is the file for which the current query performs type inference.
+    /// It acts as a token of prove that we aren't accessing an AST node from a different file
+    /// than in which the current enclosing Salsa query (which would lead to cross-file dependencies).
```

Comment:
```suggestion
    /// It acts as a token of proof that we aren't accessing an AST node from a different file
    /// to the one from which the current enclosing Salsa query is being called.
    /// Doing so would lead to cross-file dependencies, hurting incremental computation.
```